eyelid-match-ops/src/primitives/poly/modular_poly.rs

```diff
@@ -79,7 +79,9 @@ impl<C: PolyConf> Poly<C> {
     /// Converts the `coeffs` vector into a dense polynomial.
     pub fn from_coefficients_vec(coeffs: Vec<C::Coeff>) -> Self {
         let mut poly = Self(DensePolynomial { coeffs }, PhantomData);
-        poly.truncate_to_canonical_form();
+
+        poly.reduce_mod_poly();
```

Comment:
We could, but it’s already documented as an invariant of the struct:
https://github.com/Inversed-Tech/eyelid/blob/d942cc702d873d3b727191c7514cb433868df28a/eyelid-match-ops/src/primitives/poly/modular_poly.rs#L37-L66

So all public methods do this reduction (if it could possibly be needed). And the private methods which don’t do the reduction have specific docs about that.

Happy to clarify in a later PR.

Reply:
This is now in PR #83 at a struct level.